bmap-writer.cpp

```diff
@@ -26,29 +26,83 @@
 #include <string>
 #include <chrono>
 #include <cstring>
+#include <cerrno>
+
 #include <fcntl.h>
+#include <getopt.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <errno.h>
+#include <sys/sysinfo.h>
+
 #include <libxml/parser.h>
 #include <libxml/tree.h>
 #include <openssl/evp.h>
 #include <archive.h>
 
-#define CHECKSUM_LENGTH 64
-#define RANGE_LENGTH    19
-
 struct range_t {
     std::string checksum;
-    std::string range;
+    size_t startBlock;
+    size_t endBlock;
 };
 
 struct bmap_t {
     std::vector<range_t> ranges;
     size_t blockSize;
 };
 
+enum ChecksumMode {
```

Comment:
I find the checksum refactoring too much ? what kind of benefits does does bring ?

Reply:
Configurability!

Nevertheless, I got your point on the tool being minimalistic and remove this completely.